warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -319,51 +336,64 @@ private Set<Excerpt> getExcerpts(PhraseIndexes phraseIndexes) {
      * @return the excerpt
      */
     private String getExcerpt(String field, int start, int end, Range range, ArrayList<String> hitTermValues) {
+        if (start < 0) {
+            start = 0;
+        }
+        float origHalfSize = (float) (end - start) / 2;
+        int expandSize = 20;
         excerptIteratorOptions.put(TermFrequencyExcerptIterator.FIELD_NAME, field);
-        excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(start));
-        excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end));
-        try {
-            excerptIterator.init(source, excerptIteratorOptions, env);
-            excerptIterator.seek(range, Collections.emptyList(), false);
-            if (excerptIterator.hasTop()) {
-                Key topKey = excerptIterator.getTopKey();
-                String[] parts = topKey.getColumnQualifier().toString().split(Constants.NULL);
-                // The column qualifier is expected to be field\0phrase.
-                if (parts.length == 2) {
-                    return getHitPhrase(hitTermValues, parts);
+        int i = 0;
+        while (i <= 1) {
```

Comment:
If I'm reading this correctly it's possible to run the offset iterator twice in a row. Is that what's intended here?

Typically a while loop is only used when configuring scan column families or ranges in a batch scanner. It's a bit odd to see the whole iterator execution inside a while loop.

Reply:
yep. If we hit a certain condition while the iterator is running the first time, we exit out and run the iterator with a greater range the second time. Would it be more proper to put it in a for loop instead of a while loop?

Reply:
A for loop might make it a little clearer that you expect only two attempts _and_ handle the incrementing of the variable.

```
for (int i=0; i < 2; i++) { //make two attempts to do X
```